Detectors/TRD/base/include/TRDBase/Digit.h

```diff
@@ -30,36 +30,81 @@ namespace trd
 using ADC_t = std::uint16_t;
 using ArrayADC = std::array<ADC_t, constants::TIMEBINS>;
 
-using TimeStamp = o2::dataformats::TimeStamp<double>;
-
-class Digit : public TimeStamp
+// Digit class for TRD
+// Notes:
+//    Shared pads:
+//        the lowe mcm and rob is chosen for a given shared pad.
+//        this negates the need for need alternate indexing strategies.
+//        it does however mean that if you are trying to go from pad/row to mcm/rob you need to remember to manually do the shared ones.
+//        TODO we could change the get methods to return the value in negaitve to indicate a shared pad, but I feel this is just added complexity? Comments?
+//        if you are going to have to check for negative you may as well check for it being shared.
+class Digit
 {
  public:
   Digit() = default;
   ~Digit() = default;
-  Digit(const int det, const int row, const int pad, const ArrayADC adc, double t)
-    : mDetector(det), mRow(row), mPad(pad), mADC(adc), TimeStamp(t) {}
+  Digit(const int det, const int row, const int pad, const ArrayADC adc)
+  {
+    setDetector(det);
+    setRob(row, pad);
+    setMCM(row, pad);
+    setADC(adc);
+  }
+  Digit(const int det, const int row, const int pad) // add adc data in a seperate step
+  {
+    setDetector(det);
+    setRob(row, pad);
+    setMCM(row, pad);
+  }
+  Digit(const int det, const int rob, const int mcm, const int channel, const ArrayADC adc)
+  {
+    setRob(rob);
+    setMCM(mcm);
+    setChannel(channel);
+    setADC(adc);
+  }
+  Digit(const int det, const int rob, const int mcm, const int channel) // add adc data in a seperate step
+  {
+    setRob(rob);
+    setMCM(mcm);
+    setChannel(channel);
+  }
   // Copy
   Digit(const Digit&) = default;
   // Assignment
   Digit& operator=(const Digit&) = default;
   // Modifiers
+  void setRob(int rob) { mRob = rob; }
+  void setRob(int row, int pad) { mRob = FeeParam::getROBfromPad(row, pad); }
+  void setMCM(int mcm) { mMCM = mcm; }
+  void setMCM(int row, int pad) { mMCM = FeeParam::getMCMfromPad(row, pad); }
+  void setChannel(int channel) { mChannel = channel; }
   void setDetector(int det) { mDetector = det; }
-  void setRow(int row) { mRow = row; }
-  void setPad(int pad) { mPad = pad; }
   void setADC(ArrayADC const& adc) { mADC = adc; }
   // Get methods
-  int getDetector() const { return mDetector; }
-  int getRow() const { return mRow; }
-  int getPad() const { return mPad; }
+  int getDetector() const { return mRob; }
+  int getRow() const { return FeeParam::getPadRowFromMCM(mRob, mMCM); }
+  int getPad() const { return FeeParam::getPadColFromADC(mRob, mMCM, mChannel); }
+  int getRob() const { return mRob; }
+  int getMCM() const { return mMCM; }
+  int getChannel() const { return mChannel; }
+  int isSharedDigit()
+  {
+    if (mChannel == 0 || mChannel == 18 || mChannel == 19)
```

Comment:
I think this should be 0, 19 or 20.

Reply:
Wow where did i get that from..... fixed.